### PR TITLE
Make Cascade trait to enable MockCascade

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Based on https://github.com/holochain/holochain/pull/2059

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_cascade/src/lib.rs [hunk 1]
```diff
@@ -223,7 +226,145 @@ impl Cascade<HolochainP2pDna> {
     }
 }
 
-impl<Network> Cascade<Network>
+/// TODO
+#[async_trait::async_trait]
+#[mockall::automock]
+pub trait Cascade {
+    /// Retrieve [`Entry`] from either locally or from an authority.
```
Comment (jost-s) [inline crates/holochain_cascade/src/lib.rs:232]:
```suggestion
    /// Retrieve [`Entry`] either locally or from an authority.
```

Comment (jost-s) [reply crates/holochain_cascade/src/lib.rs:232]: Maybe that only sounds funny to me, it's used like this in the following comments too.

File: crates/holochain_cascade/src/lib.rs [hunk 2]
```diff
@@ -1084,3 +1116,76 @@ where
...
+                }
+                items
+            })
+            .collect();
+
+        let map0 = ShareOpen::new(map);
+
+        let map = map0.clone();
+        cascade.expect_retrieve().returning(move |hash, _| {
+            box_fut_plain(Ok(map.share_ref(|m| {
```
Comment (jost-s) [inline crates/holochain_cascade/src/lib.rs:1139]: "fut"?

**Review by jost-s — APPROVED** [review]
Great mockery!
Would we want to have a mock method that returns items as if from the network?

File: crates/holochain_cascade/src/lib.rs (hunk 2 above)
Comment (maackle) [reply crates/holochain_cascade/src/lib.rs:1139]: "future", this is meant to reduce boilerplate so I chose a short name

**Pull request merged**